src/core/crypto/sha256_mac.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2003-2024 Rony Shapiro <ronys@pwsafe.org>.
+ * All rights reserved. Use of the code is allowed under the
+ * Artistic License 2.0 terms, as specified in the LICENSE file
+ * distributed with this code, or available from
+ * http://www.opensource.org/licenses/artistic-license-2.0.php
+ */
+#include "sha256.h"
+
+/*
+  Initialize the hash state
+*/
+SHA256::SHA256()
+{
+  CC_SHA256_Init(&ctx);
+}
+
+SHA256::~SHA256()
+{
+  // CC_SHA256_Final has erased ctx
+}
+
+/*
+  Process a block of memory though the hash
+  @param in     The data to hash
+  @param inlen  The length of the data (octets)
+*/
+void SHA256::Update(const unsigned char *in, size_t inlen)
+{
+  while (inlen) {
+    unsigned int len = inlen > UINT32_MAX ? UINT32_MAX : (unsigned int) inlen;
```

Comment:
I think that this will silently break compatibility if inlen > UINT32_MAX.
Better to fail hard, I think.

Reply:
I don't understand. I added this little piece of code specifically for the case that inlen > UINT32_MAX, as the CC_SHA256_Update accepts 32-bit argument only. So this loop will feed blobs of size UINT32_MAX if inlen > UINT32_MAX.

In what sense could this break?

Reply:
My mistake. The code is fine. Apologies.